polyfill/lib/datetime.mjs

```diff
@@ -382,13 +386,19 @@ export class DateTime {
     ({ year, month, day } = ES.BalanceDate(year, month, day));
 
     const TemporalDate = GetIntrinsic('%Temporal.Date%');
-    const adjustedLarger = new TemporalDate(year, month, day, calendar);
+    const adjustedDate = new TemporalDate(year, month, day, calendar);
+    const otherDate = new TemporalDate(
+      GetSlot(other, ISO_YEAR),
+      GetSlot(other, ISO_MONTH),
+      GetSlot(other, ISO_DAY),
+      calendar
+    );
```

Comment:
It was because ES.DifferenceDate now calls Temporal.Date.compare on the dates, which is branded. But I should not be doing that, anyway, so I'll revise this.

Reply:
Actually, there is a bit more surgery to do on the spec. It seems that abstract operations like CompareTemporalDate and CompareTemporalTime should not have brand checks, only the methods themselves should. I'd prefer to file a separate issue for this.

Reply:
→ #864